File: tests/correct_test/correct_all.pas
program ExtendedPascalTest;

var
  str1, str2: string;
  ch: char;
  i: integer;

function GetNumber: integer;
begin
  GetNumber := 42;
end;

begin
  { 字符串操作测试 }
  str1 := 'Hello';
  ch := '!';
  write(str1);
  
  { switch-case语句测试 }
  i := 2;
  case i of
    1: write('One');
    2: begin
         write('Two');
         break;  { break语句测试 }
       end;
    3: write('Three');
  end;
  
  { continue语句测试 }
  for i := 1 to 5 do
  begin
    if i = 3 then continue;
    write(i);
  end;
end.

File: tests/correct_test/correct_all.c
#include <stdio.h>

const char* str1;
const char* str2;
char ch;
int i;
int getnumber() {
    int getnumber_result;
    getnumber_result = 42;
    return getnumber_result;
}


int main() {
    str1 = "Hello";
    ch = '!';
    printf("%s", str1);
    i = 2;
    switch (i) {
        case 1:
            printf("%s", "One");
            break;
        case 2:
            {
                printf("%s", "Two");
                break;
            }
            break;
        case 3:
            printf("%s", "Three");
            break;
    }
    for (i = 1; i <= 5; i++) {
        if ((i == 3)) {
            continue;
        }
        printf("%d", i);
    }
    return 0;
}
